library/ssl_tls.c

```diff
@@ -2384,15 +2455,13 @@ int mbedtls_ssl_set_hostname( mbedtls_ssl_context *ssl, const char *hostname )
 
     /* Now it's clear that we will overwrite the old hostname,
      * so we can free it safely */
-
```

Comment:
restore it. I do not think we should fix it in this PR. 

But for `mbedtls_ssl_session_set_hostname` , I think we should fix it.